lib/modules/blockchain_connector/provider.js

```diff
@@ -72,15 +72,26 @@ class Provider {
     });
   }
 
+  connected() {
+    if (this.type === 'rpc') {
+      return !!this.provider;
+    } else if (this.type === 'ws') {
+      return this.provider && this.provider.connected;
+    }
```

Comment:
There is also the `vm` type that was added with the test refactor.

Reply:
Good spot, I guess VM is the same as rpc?

Reply:
It's a Ganache provider, but yeah, same as RPC would work.